news/_posts/2018-03-15-exposing-data-with-confidence.md

```diff
@@ -0,0 +1,137 @@
+
+In comparison, a whitelist allows shops to be **safe by default**. Models can evolve in any way, and data won't accidentally be leaked – by design. Developers need to _add_ new information into the whitelist if they want to expose it. If it's not used, everything will work fine. But if it is used, they'll notice it right away.
+
+## So how does this impact me?
+
+If you have customized your PrestaShop or if you are a module developer, here's what you need to know.
+
+### I customized my models but I only use my custom data in Smarty templates
+
+Good news! You don't need to do anything. As stated before, objects being sent to the templates are not filtered, so everything will work the same as before.
+
+### I customized my models and I need my custom data on the client side
+
+You need to add your data keys to the front end filter service. Here's how.
+
+Since PrestaShop 1.7.3, modules are able to access the dependency container in back office controllers. Starting on 1.7.3.1, they will also have access to the front office dependency container, so accessing the filter service from your module is as simple as:
+
+```php
+$this->get('prestashop.core.filter.front_end_object.main');
+```
+
+#### Some examples
+
+##### Whitelisting a property
+
+Let's say you have added `puffin_product_url` to your `Product` model and you want it to be made available on the browser when the customer adds a product to his cart.
+
+Here's how you can add `puffin_product_url` to the whitelist by using a module attached to the `actionFrontControllerAfterInit` hook:
+
+```php
+public function hookActionFrontControllerAfterInit()
+{
+    $filterManager = $this->get('prestashop.core.filter.front_end_object.main');
+    
+    // get list of all filters applied to client-side data
+    $filters = $filterManager->getFilters();
+    
+    // get list of filters applied to the cart object
+    $cartFilters = $filters['cart']->getFilters();
+    
+    // add puffin_product_url to the product whitelist (remember to pass an array)
+    $cartFilters['products']->whitelist(['puffin_product_url']);
+}
+```
+
+##### Removing a property
+
+Let's say you want to remove `price` from your exposed cart Products.
+
+Again, you can hook into `actionFrontControllerAfterInit` and remove `price` from the product whitelist:
+
+```php
+public function hookActionFrontControllerAfterInit()
+{
+    $filterManager = $this->get('prestashop.core.filter.front_end_object.main');
+    
+    // get list of all filters applied to client-side data
+    $filters = $filterManager->getFilters();
+    
+    // get list of filters applied to the cart object
+    $cartFilters = $filters['cart']->getFilters();
+    
+    // remove "price" from the product whitelist
+    $cartFilters['products']->removeFromWhitelist('price');
+}
+```
+
+##### Bonus: adding arbitrary data to the prestashop object
```

Comment:
how about: "Bonus: make new information available in Front Office"

Reply:
Why not!